### ISSUE TITLE
Review missingFileHandler:  and installed: behaviour

### ISSUE THREAD

**Issue by naseemkullah** [roboll/helmfile#519]
Hello,

I would like to use `missingFileHandler: Error`,

However I have various environments declared, but not all environments have all charts installed. E.g:

```
environments:
- env1
- env2
- env3
- env4

releases:
- name: elasticsearch
  chart: stable/elasticsearch
  installed: {{ eq .Environment.Name "env1" "env4" }}
  values:
  - config/{{`{{ .Release.Name }}`}}/{{`{{ .Environment.Name }}`}}.yaml

- name: nginx-ingress
  chart: stable/nginx-ingress
  installed: {{ eq .Environment.Name "env2" "env3" }}
  values:
  - config/{{`{{ .Release.Name }}`}}/{{`{{ .Environment.Name }}`}}.yaml
```

This will result in an error if using `missingFileHandler: Error`
Naturally I do not have values files for `nginx-ingress` for environments `env1` and `env4`.
Nor do I have values files for `elasticsearch` for environments `env2` and `env3`

But I would still like to use `missingFileHandler: Error`

TL;DR, could `missingFileHandler: Error` only check if files exist for releases that will be installed? not check for those that are not intended to be installed (or intended to be deleted) ?

Many thanks!

**Comment by mumoshu** [comment]
Hey! Thanks for writing up the detailed explanation
of your issue.
This seems like a valid enhancement to helmfile.
Would you mind submitting a PR?
2019年3月30日(土) 6:43 Naseem <notifications@github.com>:

**Comment by naseemkullah** [comment]
Hi @mumoshu thanks for confirming that this is a valid enhancement!
I am still a beginner at Go but will have a go at it (pun intended)!

**Comment by naseemkullah** [comment]
I've noticed that the behaviour is only when using `sync`, but with `apply` the behaviour is correct (only errors out if release that should be installed is missing file)

**Comment by mumoshu** [comment]
@naseemkullah Thanks! Your additional report was helpful! I managed to fix it in #526.

**Comment by naseemkullah** [comment]
Thank you @mumoshu !

**Comment by mumoshu** [comment]
The fix is included in v0.51.1 released just now
